### ISSUE TITLE
[BUG] Index FILTER expression does not work for nested JSON values

### ISSUE THREAD

**Issue by blendicavlad** [RediSearch/RediSearch#4898]
**Describe the bug**
Adding a FILTER to a JSON Index that filters based on a nested field filters out the entire dataset 

**To Reproduce**
1. create the following index: `FT.CREATE testIndex ON JSON PREFIX 1 'testData' FILTER "@details\\.status==A" SCHEMA TAG $.job as job TAG $.details.status AS details.status TAG`
2. add data: `json.set testData.1 . "{\"job\":\"SRE\",\"details\":{\"status\":\"A\"}}"` `json.set testData.2 . "{\"job\":\"AAA\",\"details\":{\"status\":\"D\"}}"`

**Expected behavior**
Only testData.1 key should be indexed, but both keys are indexed.

**Environment (please complete the following information):**
 - OS: RedHat 8.8 Ootpa
 - CPU model: Intel Xeon Gold 5317
 - Version/branch: Redis 7.2.3, RedisSearch 2.8.9

**Additional context**
If i set the filter as "@job==SRE" (a top level field) then it works as expected. I couldn't find anything about JSON Path support or nested fields behaviour on docs for Redis Aggregation Language.

**Comment by GuyAv46** [comment]
Hey @blendicavlad, thanks for reaching out.

The main issue is that field names should not have punctuation in them, as the query's (and filter/apply steps') parser will misinterpret the name and the query will result in a syntax error (as you experienced yourself).

Attempting to define the index using an underscore instead of a dot seems to work:

```
127.0.0.1:6379> FT.CREATE testIndex ON JSON PREFIX 1 'testData' FILTER "@details_status=='A'" SCHEMA $.job as job TAG $.details.status AS details_status TAG
OK
127.0.0.1:6379> json.set testData.1 . "{\"job\":\"SRE\",\"details\":{\"status\":\"A\"}}"
OK
127.0.0.1:6379> json.set testData.2 . "{\"job\":\"AAA\",\"details\":{\"status\":\"D\"}}"
OK
127.0.0.1:6379> FT.SEARCH testIndex *
1) (integer) 1
2) "testData.1"
3) 1) "$"
   2) "{\"job\":\"SRE\",\"details\":{\"status\":\"A\"}}"
```

Let me know if there's anything else I can help with

**Comment by blendicavlad** [comment]
Yeah, as a workaround this seems to work, thank you! It would be great if filter expressions could be escaped similar to FT queries, e.g. @parent\\.child:ABC. Is there currently any parsing limitation of why this isn't the case for FILTER/APPLY expressions as well? If not, I can try to work on a PR for it

**Comment by GuyAv46** [comment]
The `FILTER/APPLY` expression parser differs from the query parser. The issue is backward compatibility - we need to see that the fix will not ruin working cases that some users might use. You're welcome to try and implement it.

We plan to refactor this parser in the next major release. I'll make sure we address this case when we do

**Comment by blendicavlad** [comment]
Thats great to hear then. Should I close this issue or relabel it as feature?

**Comment by GuyAv46** [comment]
Lets close and open a new one ("support punctuation in field name on `FILTER`/`APPLY`" or something similar)
